LoginViewController 코드 분리 및 앱 백그라운드 설정

song1900 opened the pull request:
- LoginViewController에서 UI 관련 코드 분리
- 앱 백그라운드 컬러 설정

song1900 committed:
Style: 앱 전체의 배경색 설정

song1900 committed:
Refactor: LoginViewController 에서 UI 관련 코드 분리 작업

song1900 committed:
Merge branch 'main' into feature/separate-login-view

f-lab-jercy commented on GIL/GIL/Screen/Login/LoginView.swift:
```diff
@@ -0,0 +1,83 @@
+    }
+    
+    
+    func makeConstraints() {
+        emailTextField.topAnchor.constraint(equalTo: topAnchor, constant: 120).isActive = true
+        emailTextField.leadingAnchor.constraint(equalTo: leadingAnchor, constant: 16).isActive = true
+        emailTextField.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -16).isActive = true
+        emailTextField.heightAnchor.constraint(equalToConstant: 55).isActive = true
+        
+        passwordTextField.topAnchor.constraint(equalTo: emailTextField.bottomAnchor, constant: 10).isActive = true
+        passwordTextField.leadingAnchor.constraint(equalTo: emailTextField.leadingAnchor).isActive = true
+        passwordTextField.trailingAnchor.constraint(equalTo: emailTextField.trailingAnchor).isActive = true
+        passwordTextField.heightAnchor.constraint(equalTo: emailTextField.heightAnchor).isActive = true
+        
+        loginButton.topAnchor.constraint(equalTo: passwordTextField.bottomAnchor, constant: 30).isActive = true
+        loginButton.leadingAnchor.constraint(equalTo: passwordTextField.leadingAnchor).isActive = true
+        loginButton.trailingAnchor.constraint(equalTo: passwordTextField.trailingAnchor).isActive = true
+        loginButton.heightAnchor.constraint(equalToConstant: 60).isActive = true
+        
+        stackView.topAnchor.constraint(equalTo: loginButton.bottomAnchor, constant: 40).isActive = true
+        stackView.centerXAnchor.constraint(equalTo: centerXAnchor).isActive = true
+    }
```
오토레이아웃 잡는 코드는 updateConstraints에 추가를 해도 좋을것 같습니다.
그리고 isactive를 매번 하는것보단 NSConstraints.activate와 같은 메서드를 이용하면 더 깔끔할것 같아보여요

f-lab-jercy commented on GIL/GIL/Screen/Login/LoginViewController.swift:
```diff
@@ -9,113 +9,63 @@ import UIKit
+    }
+    
+    private func bindButtons() {
+        loginView.loginButton.addTarget(self, action: #selector(loginButtonTapped), for: .touchUpInside)
+        loginView.signUpButton.addTarget(self, action: #selector(signUpButtonTapped), for: .touchUpInside)
```
objc메서드를 사용하지 않게 새로나온 버튼 액션을 이용해보면 좋을것 같습니다.

f-lab-jercy commented on GIL/GIL/Screen/Login/LoginViewController.swift:
```diff
@@ -9,113 +9,63 @@ import UIKit
 
 class LoginViewController: UIViewController {
```
가능하다면 final class로 상속받지 않는 클래스라고 명시해주면 좋을것 같습니다.

song1900 committed:
Merge branch 'main' of https://github.com/f-lab-edu/gil-navi-ios into feature/separate-login-view

# Conflicts:
#	GIL/GIL.xcodeproj/project.pbxproj

song1900 committed:
Chore: 기존 파일들 제거

song1900 committed:
Refactor: 오토레이아웃 잡는 코드 수정

- Drawing Cycle updateConstraints 메서드로 이동
- NSLayoutConstraoint.activate 사용해 한 번에 constraint를 설정

song1900 committed:
Merge branch 'feature/separate-login-view' of https://github.com/f-lab-edu/gil-navi-ios into feature/separate-login-view

song1900 committed:
Chore: 상속받지 않는 LoginViewController Class에 final 키워드 추가

song1900 committed:
Refactor: UIButton Action 추가 방법 수정

- addTarget으로 추가하는 방법에서 UIAction을 추가하는 방법으로 수정
